code/game/machinery/virtual_reality/vr_console.dm

```diff
@@ -1,227 +1,260 @@
-
-	else
-		occupant.enter_vr(avatar)
-
+/obj/machinery/vr_sleeper
+	name = "virtual reality sleeper"
+	desc = "A fancy bed with built-in sensory I/O ports and connectors to interface users' minds with their bodies in virtual reality."
+	icon = 'icons/obj/Cryogenic2.dmi'
+	icon_state = "syndipod_0"
+
+	var/base_state = "syndipod_"
+
+	density = 1
+	anchored = 1
+	circuit = /obj/item/weapon/circuitboard/vr_sleeper
+	var/mob/living/carbon/human/occupant = null
+	var/mob/living/carbon/human/avatar = null
+	var/datum/mind/vr_mind = null
+	var/datum/effect/effect/system/smoke_spread/bad/smoke
+
+	var/eject_dead = TRUE
+
+	use_power = 1
+	idle_power_usage = 15
+	active_power_usage = 200
+	light_color = "#FF0000"
+
+/obj/machinery/vr_sleeper/New()
+	..()
+	component_parts = list()
+	component_parts += new /obj/item/weapon/stock_parts/scanning_module(src)
+	component_parts += new /obj/item/stack/material/glass/reinforced(src, 2)
+
+	RefreshParts()
+
+/obj/machinery/vr_sleeper/Initialize()
+	. = ..()
+	smoke = new
+	update_icon()
+
+/obj/machinery/vr_sleeper/Destroy()
+	. = ..()
+	go_out()
+
+/obj/machinery/vr_sleeper/process()
+	if(stat & (NOPOWER|BROKEN))
+		if(occupant)
+			go_out()
+			visible_message("<span class='notice'>\The [src] emits a low droning sound, before the pod door clicks open.</span>")
+		return
+	else if(eject_dead && occupant && occupant.stat == DEAD) // If someone dies somehow while inside, spit them out.
+		visible_message("<span class='warning'>\The [src] sounds an alarm, swinging its hatch open.</span>")
+		go_out()
+
+/obj/machinery/vr_sleeper/update_icon()
+	icon_state = "[base_state][occupant ? "1" : "0"]"
+
+/obj/machinery/vr_sleeper/Topic(href, href_list)
+	if(..())
+		return 1
+
+	if(usr == occupant)
+		to_chat(usr, "<span class='warning'>You can't reach the controls from the inside.</span>")
+		return
+
+	add_fingerprint(usr)
+
+	if(href_list["eject"])
+		go_out()
+
+	return 1
+
+/obj/machinery/vr_sleeper/attackby(var/obj/item/I, var/mob/user)
+	add_fingerprint(user)
+
+	if(occupant && (istype(I, /obj/item/device/healthanalyzer) || istype(I, /obj/item/device/robotanalyzer)))
+		I.attack(occupant, user)
+		return
+
+	if(default_deconstruction_screwdriver(user, I))
+		return
+	else if(default_deconstruction_crowbar(user, I))
+		if(occupant && avatar)
+			avatar.exit_vr()
+			avatar = null
+			go_out()
+		return
+
+
+/obj/machinery/vr_sleeper/MouseDrop_T(var/mob/target, var/mob/user)
+	if(user.stat || user.lying || !Adjacent(user) || !target.Adjacent(user)|| !isliving(target))
+		return
+	go_in(target, user)
+
+
+
+/obj/machinery/sleeper/relaymove(var/mob/user)
+	..()
+	if(usr.incapacitated())
+		return
+	go_out()
+
+
+
+/obj/machinery/vr_sleeper/emp_act(var/severity)
+	if(stat & (BROKEN|NOPOWER))
+		..(severity)
+		return
+
+	if(occupant)
+		// This will eject the user from VR
+		// ### Fry the brain? Yes. Maybe.
+		if(prob(severity * 15) && occupant.species.has_organ[O_BRAIN] && occupant.internal_organs_by_name[O_BRAIN])
```

Comment:
Looks like this severity thing might be backwards? High Severity means a less powerful EMP pulse. Severity = 1 is the strongest it can be.
As written this makes minor EMPs 4 times as likely and 4 times as damaging as a direct hit would be.